ironfish/src/blockchain/blockchain.ts

```diff
@@ -1008,7 +1009,7 @@ export class Blockchain {
       )
 
       const block = new Block(header, transactions)
-      if (!previousBlockHash.equals(GENESIS_BLOCK_PREVIOUS)) {
+      if (!previousBlockHash.equals(GENESIS_BLOCK_PREVIOUS) && verifyBlock) {
```

Comment:
We can avoid the buffer comparison when verifyBlock is false if we move it first. Pretty minor over-optimization, but 🤷 
```suggestion
      if (verifyBlock && !previousBlockHash.equals(GENESIS_BLOCK_PREVIOUS)) {
```